### DISCUSSION TITLE
JS SDK 9 - React Native - App Check

### DISCUSSION THREAD

**Discussion by anhnch** [firebase/firebase-js-sdk#5833] [Q&A]
Hi,

I'm trying to use the Firebase JS SDK 9 to my React Native. I don't use the [rn-firebase](https://rnfirebase.io/) libraries because they are still for SDK 8 and some other reasons. Everything is fine. Now I'd like to use App Check for my app.

```typescript
import { initializeApp } from '@firebase/app';
import { getFirestore } form '@firebase/firestore';
import { initializeAppCheck, ReCaptchaV3Provider } from '@firebase/app-check';

const app = initializeApp(config, 'MyApp');

// do I have to activate the AppCheck here ?
const appCheck = initializeAppCheck(app, {
    // the JS SDK doesn't support DeviceCheck, AppAttest, SaftyNet
    // do I have to use ReCaptchaV3Provider? 
    provider: new ReCaptchaV3Provider('mykey'),
    isTokenAutoRefreshEnabled: true
})
const firestore = getFirestore(app);
```
I have tried another way
```
pod 'Firebase/AppCheck'
```
and in AppDelegate.m
```Objective-C
  FIRAppCheckDebugProviderFactory *providerFactory = [[FIRAppCheckDebugProviderFactory alloc] init];
  [FIRAppCheck setAppCheckProviderFactory:providerFactory];
  [FIRApp configure];
```
and added generated token to firebase console debug token. But it doesn't work, I got 'permission denied'. I guess the JS SDK firestore requests are not relevant to what I set in AppDelegate.m

The documents are for iOS, Android, and Web, while I'm trying to use the JS SDK (web) for React Native. I'm not sure what document should be referred to.

Has anyone worked with App Check, React Native, and Fire JS SDK 9 together?

**Comment by hsubox76** [comment]
You can maybe use the [`CustomProvider` class](https://firebase.google.com/docs/reference/js/app-check.customprovider) (instead of ReCaptchaV3Provider) and provide your own implementation of `getToken()` that somehow calls DeviceCheck/AppAttest/SafetyNet, if you're able to write the code for that. That doesn't seem trivial, but as the JS SDK is a web SDK, it doesn't seem practical for us to provide built-in code that only works on mobile devices.

**Reply by anhnch** [reply]
Thank you for your advice
